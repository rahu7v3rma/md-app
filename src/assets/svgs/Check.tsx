import React, { FunctionComponent } from 'react';
import { Path, Rect, Svg } from 'react-native-svg';

import { Colors } from '@/theme/colors';

type Props = {
    width?: number;
    height?: number;
    color?: string;
};

const CheckIcon: FunctionComponent<Props> = ({
    width = 97,
    height = 96,
    color = Colors.extras.white
}: Props) => {
    return (
        <Svg width={width} height={height} viewBox="0 0 97 96" fill="none">
            <Rect
                x={0.5}
                width={width}
                height={height}
                rx={32}
                fill={Colors.theme.primary}
            />
            <Path
                d="M62.5 38l-20 20-8-8"
                stroke={color}
                strokeWidth={4}
                strokeLinecap="round"
                strokeLinejoin="round"
            />
        </Svg>
    );
};

export default CheckIcon;
