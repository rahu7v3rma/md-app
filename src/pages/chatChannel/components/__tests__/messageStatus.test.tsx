import React from 'react';
import { create } from 'react-test-renderer';
import {
    isMessageWithStylesReadByAndDateSeparator,
    useMessageContext
} from 'stream-chat-react-native';

import Check from '@/assets/svg/check';
import DoubleCheck from '@/assets/svg/doubleCheck';

import MessageStatus from '../messageStatus';

jest.mock('stream-chat-react-native', () => ({
    isMessageWithStylesReadByAndDateSeparator: jest.fn(),
    useMessageContext: jest.fn()
}));

const mockMessage = ({
    withStyles,
    isMyMessage,
    type,
    readBy,
    status
}: any) => {
    (
        isMessageWithStylesReadByAndDateSeparator as unknown as jest.Mock
    ).mockReturnValueOnce(withStyles);
    (useMessageContext as jest.Mock).mockReturnValueOnce({
        isMyMessage,
        message: { type, readBy, status }
    });
};

test('Match Snapshot - Check', () => {
    mockMessage({
        withStyles: false,
        isMyMessage: true,
        type: 'regular',
        readBy: false,
        status: 'received'
    });
    expect(create(<MessageStatus />).toJSON()).toMatchSnapshot();
});

test('Match Snapshot - DoubleCheck', () => {
    mockMessage({
        withStyles: true,
        isMyMessage: true,
        type: 'regular',
        readBy: true,
        status: 'received'
    });
    expect(create(<MessageStatus />).toJSON()).toMatchSnapshot();
});

test('renders without crashing', () => {
    mockMessage({
        withStyles: true,
        isMyMessage: true,
        type: 'regular',
        readBy: true,
        status: 'received'
    });
    const tree = create(<MessageStatus />);
    expect(tree).toBeTruthy();
});

test('renders DoubleCheck component when the message is read', () => {
    mockMessage({
        withStyles: true,
        isMyMessage: true,
        type: 'regular',
        readBy: true,
        status: 'received'
    });
    const tree = create(<MessageStatus />);
    expect(tree.root.findAllByType(DoubleCheck)).toHaveLength(1);
});

test('renders Check component when the message status is received', () => {
    mockMessage({
        withStyles: false,
        isMyMessage: true,
        type: 'regular',
        readBy: false,
        status: 'received'
    });
    const tree = create(<MessageStatus />);
    expect(tree.root.findAllByType(Check)).toHaveLength(1);
});

test('does not render any component when the message is not from the current user', () => {
    mockMessage({
        withStyles: true,
        isMyMessage: false,
        type: 'regular',
        readBy: true,
        status: 'received'
    });
    const tree = create(<MessageStatus />);
    expect(tree.root.children).toHaveLength(0);
});

test('does not render any component when the message type is error', () => {
    mockMessage({
        withStyles: true,
        isMyMessage: true,
        type: 'error',
        readBy: true,
        status: 'received'
    });
    const tree = create(<MessageStatus />);
    expect(tree.root.children).toHaveLength(0);
});

test('does not render any component when the message readBy property is not a number', () => {
    mockMessage({
        withStyles: true,
        isMyMessage: true,
        type: 'regular',
        readBy: 'readBy',
        status: 'notReceived'
    });
    const tree = create(<MessageStatus />);
    expect(tree.root.children).toHaveLength(0);
});

test('does not render any component when the message readBy property is not true', () => {
    mockMessage({
        withStyles: true,
        isMyMessage: true,
        type: 'regular',
        readBy: false,
        status: 'notReceived'
    });
    const tree = create(<MessageStatus />);
    expect(tree.root.children).toHaveLength(0);
});
