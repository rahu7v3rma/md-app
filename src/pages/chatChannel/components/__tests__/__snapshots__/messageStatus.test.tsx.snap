// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot - Check 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={8}
  bbWidth={9}
  fill="none"
  focusable={false}
  height={8}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 8,
        "width": 9,
      },
    ]
  }
  vbHeight={8}
  vbWidth={9}
  width={9}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M8.247 1.173a.601.601 0 00-.849-.007L2.963 5.549 1.035 3.553a.601.601 0 10-.868.833l.004.003 2.35 2.436c.113.115.267.18.427.18h.007a.601.601 0 00.42-.173l4.866-4.81a.601.601 0 00.006-.85z"
      fill={4286873440}
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`Match Snapshot - DoubleCheck 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={12}
  bbWidth={12}
  fill="none"
  focusable={false}
  height={12}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 12,
        "width": 12,
      },
    ]
  }
  vbHeight={12}
  vbWidth={12}
  width={12}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGGroup
      clipPath="clip0_9_258"
      fill={4286873440}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M8.247 3.173a.601.601 0 00-.849-.007L2.963 7.55 1.035 5.553a.601.601 0 10-.868.833l.004.003 2.35 2.436c.113.115.267.18.427.18h.007a.601.601 0 00.42-.173l4.866-4.81a.601.601 0 00.006-.85zM6.13 8.825c.112.115.265.18.425.18h.007a.601.601 0 00.42-.173l4.866-4.81a.601.601 0 00-.842-.856L6.573 7.55l-.158-.164a.601.601 0 10-.868.833l.003.003.58.604z"
      />
    </RNSVGGroup>
    <RNSVGDefs>
      <RNSVGClipPath
        name="clip0_9_258"
      >
        <RNSVGPath
          d="M0 0H12V12H0z"
          fill={4294967295}
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGClipPath>
    </RNSVGDefs>
  </RNSVGGroup>
</RNSVGSvgView>
`;
